src/storage/FileResourceMapper.ts

```diff
@@ -0,0 +1,48 @@
+import { types } from 'mime-types';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { NotFoundHttpError } from '../util/errors/NotFoundHttpError';
+import { FileResourceStore } from './FileResourceStore';
+import { ResourceMapper } from './ResourceMapper';
+
+export class FileResourceMapper implements ResourceMapper {
+  private readonly fileStore: FileResourceStore;
+  private readonly types: Record<string, any>;
+
+  public constructor(fileStore: FileResourceStore, overrideTypes = { acl: 'text/turtle', metadata: 'text/turtle' }) {
+    this.fileStore = fileStore;
+    this.types = { ...types, ...overrideTypes };
+  }
+
+  /**
+   * Strips the baseRequestURI from the identifier and checks if the stripped base URI matches the store's one.
+   * @param identifier - Incoming identifier.
+   *
+   * @throws {@link NotFoundHttpError}
+   * If the identifier does not match the baseRequestURI path of the store.
+   */
+  public mapUrlToFilePath(identifier: ResourceIdentifier): string {
+    if (!identifier.path.startsWith(this.fileStore.baseRequestURI)) {
+      throw new NotFoundHttpError();
+    }
+    return identifier.path.slice(this.fileStore.baseRequestURI.length);
+  }
+
+  /**
+   * Strips the rootFilepath path from the filepath and adds the baseRequestURI in front of it.
+   * @param path - The filepath.
+   *
+   * @throws {@Link Error}
+   * If the filepath does not match the rootFilepath path of the store.
+   */
+  public mapFilePathToUrl(path: string): string {
+    if (!path.startsWith(this.fileStore.rootFilepath)) {
+      throw new Error(`File ${path} is not part of the file storage at ${this.fileStore.rootFilepath}.`);
+    }
+    return this.fileStore.baseRequestURI + path.slice(this.fileStore.rootFilepath.length);
```

Comment:
I'm always hesitant when I see URLs being created like this since I'm never quite sure where the slashes are going to be, so this might be a good time to start using a library like `url-join` to join the parts (since the URL library of node can sometimes do some weird things when joining).

Reply:
This one should be safe actually (because we know what both parts look like), but we want to ensure (percent)-escaping. See NSS tests.

Reply:
I'm still in favour of it because it clearly shows what we're doing and what the parts correspond to. I remember in the previous solid store there were some issues because `path.join` got used for both file paths and urls (causing issues on windows for example), so I think it would be cleaner to always use either an url join or path join when doing things like this.

Reply:
But double escaping is also trouble.

The important thing is that the unit tests cover it. I don't mind how it's implemented.